src/resources/formats/html/bootstrap/_bootstrap-rules.scss

```diff
@@ -907,6 +907,12 @@ pre {
 // Maps the pandoc 'monobackgroundcolor' to bootstrap
 // Note this only targets code outside of sourceCode blocks
 @if variable-exists(mono-background-color) {
```

Comment:
Looking at this, I am wondering if this part of the code (the oldest from https://github.com/mcanouil/quarto-cli/commit/59591f61a705414c398f0fd96d003132ae193722) is not unnecessary anymore. 
https://github.com/quarto-dev/quarto-cli/blob/e637c76b14e5327a4053ff0ffcaa8cab27b672dd/src/resources/formats/html/bootstrap/_bootstrap-rules.scss#L907-L924
It seems we are duplicating it with what is below 
https://github.com/quarto-dev/quarto-cli/blob/e637c76b14e5327a4053ff0ffcaa8cab27b672dd/src/resources/formats/html/bootstrap/_bootstrap-rules.scss#L926-L949

The later is more conditional to variables than the former, and it will overwrite it too. 

Possibly that was the issue for #4746 fixed by adding a duplicated conditional in 9fff5b7cc4ece31767c91ebf5752d979d6f07999

Just sharing this observation as it could be an opportunity to simplify. We could also leverage more SASS syntax to avoid writing and maintaining the expanded version of all tags and rules with `code:not(.sourceCode)` - but that is ok like this also.

Reply:
From my few tests, I did not notice something wrong with this "duplication" and assumed it was meant to fix something elsewhere.

Reply:
Yes I think it would be the second. I believe it is duplicate because when adding the second, the first already existed. And then it was overriding the first, which caused #4746 I believe.

Reply:
Merged the two rules into one (the second): f36fb64f11140c9a32605e476ae7f5c2e2a6113f